### PR TITLE
Use network isolation parameters in docs & tests (backend + Keystone)

### PR DESCRIPTION
**Pull request by jistr**

### CODE REVIEW

File: tests/roles/backend_services/tasks/main.yaml [hunk 1]
```diff
@@ -94,25 +113,20 @@
...
-      nova:
+      placement:
         enabled: false
         template: {}
 
-      ironic:
-        enabled: false
-        template:
-          ironicConductors: []
+      rabbitmq:
```
Comment (jistr) [inline tests/roles/backend_services/tasks/main.yaml:120]: Now i see i forgot the net isolation parameters in the test suite here for backend services. The one in Keystone test seems to have worked tho.

File: docs/openstack/mariadb_copy.md [hunk 2]
```diff
@@ -33,12 +33,12 @@ PODIFIED_MARIADB_IP=$(oc get -o yaml pod mariadb-openstack | grep podIP: | awk '
 MARIADB_IMAGE=quay.io/podified-antelope-centos9/openstack-mariadb:current-podified
 
 # Use your environment's values for these:
-EXTERNAL_MARIADB_IP=192.168.24.3
+EXTERNAL_MARIADB_IP=172.21.0.2
```
Comment (jistr) [inline docs/openstack/mariadb_copy.md:36]: 172.21.0.100

Comment (jistr) [reply docs/openstack/mariadb_copy.md:36]: Actually it has to be 192.168.122.100. SSH works on the external IP too, but MariaDB only listens only on the "primary" IP of the VM.

**Comment by jistr** [comment]
Indeed the failure looks unrelated to the PR. The Wallaby Standalone fails to deploy. There is the same top-level error as before:

```
TASK [mariadb_copy : MariaDB copy pre checks] **********************************

[ABRIDGED]

 "ERROR 2002 (HY000): Can't connect to MySQL server on '38.102.83.42' (115)"], "stdout": "", "stdout_lines": []}
```

And the cause is:

```
SYNC POLL on undercloud: jid=959520998050.88662 started=1 finished=0
ASYNC POLL on undercloud: jid=959520998050.88662 started=1 finished=0
ASYNC POLL on undercloud: jid=959520998050.88662 started=1 finished=0
ASYNC POLL on undercloud: jid=959520998050.88662 started=1 finished=0
fatal: [undercloud]: FAILED! => {"ansible_job_id": "959520998050.88662", "changed": false, "cmd": "openstack tripleo deploy  --templates $DEPLOY_TEMPLATES   --yes --output-dir $DEPLOY_OUTPUT_DIR  --stack $DEPLOY_STACK --standalone-role $DEPLOY_STANDALONE_ROLE --timeout $DEPLOY_TIMEOUT_ARG -e /usr/share/openstack-tripleo-heat-templates/environments/standalone/standalone-tripleo.yaml -e /home/zuul-worker/containers-prepare-parameters.yaml -e /home/zuul-worker/standalone_parameters.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/low-memory-usage.yaml -r $DEPLOY_ROLES_FILE      --deployment-user $DEPLOY_DEPLOYMENT_USER  --local-ip $DEPLOY_LOCAL_IP --control-virtual-ip $DEPLOY_CONTROL_VIP          >/home/zuul-worker/standalone_deploy.log 2>&1", "delta": "0:25:53.755481", "end": "2023-06-06 08:34:56.056263", "finished": 1, "msg": "non-zero return code", "rc": 1, "start": "2023-06-06 08:09:02.300782", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}
```

and in standalone_deploy.log:

```
2023-06-06 08:33:44.447625 | fa163ea3-aa16-5feb-f998-000000002242 |       TASK | Run init bundle puppet on the host for mysql
2023-06-06 08:34:49.194420 | fa163ea3-aa16-5feb-f998-000000002242 |      FATAL | Run init bundle puppet on the host for mysql | standalone | error={"changed": false, "cmd": "puppet apply  --detailed-exitcodes --summarize --color=false --modulepath '/etc/puppet/modules:/opt/stack/puppet-modules:/usr/share/openstack-puppet/modules' --tags 'pacemaker::resource::bundle,pacemaker::property,pacemaker::resource::ocf,pacemaker::constraint::order,pacemaker::constraint::colocation' -e '[\"Mysql_datadir\", \"Mysql_user\", \"Mysql_database\", \"Mysql_grant\", \"Mysql_plugin\"].each |String $val| { noop_resource($val) }; include ::tripleo::profile::base::pacemaker; include ::tripleo::profile::pacemaker::database::mysql_bundle'\n", "delta": "0:01:04.489477", "end": "2023-06-06 12:34:49.152037", "failed_when_result": true, "msg": "non-zero return code", "rc": 6, "start": "2023-06-06 12:33:44.662560", "stderr": "Warning: /etc/puppet/hiera.yaml: Use of 'hiera.yaml' version 3 is deprecated. It should be converted to version 5\n   (file: /etc/puppet/hiera.yaml)\nWarning: Undefined variable '::deploy_config_name'; \n   (file & line not available)\nWarning: The function 'hiera' is deprecated in favor of using 'lookup'. See https://puppet.com/docs/puppet/7.10/deprecated_language.html\n   (file & line not available)\nDeprecation Warning: This command is deprecated and will be removed. Please use 'pcs property config' instead.\nDeprecation Warning: This command is deprecated and will be removed. Please use 'pcs constraint config' instead.\nDeprecation Warning: This command is deprecated and will be removed. Please use 'pcs constraint config' instead.\nError: pcs -f /var/lib/pacemaker/cib/puppet-cib-backup20230606-143531-1pzw1f constraint location galera-bundle rule resource-discovery=exclusive score=0 galera-role eq true failed: Error: duplicate constraint already exists, use --force to override. Too many tries\nError: /Stage[main]/Tripleo::Profile::Pacemaker::Database::Mysql_bundle/Pacemaker::Resource::Ocf[galera]/Pcmk_resource[galera]/ensure: change from 'absent' to 'present' failed: pcs -f /var/lib/pacemaker/cib/puppet-cib-backup20230606-143531-1pzw1f constraint location galera-bundle rule resource-discovery=exclusive score=0 galera-role eq true failed: Error: duplicate constraint already exists, use --force to override. Too many tries", "stderr_lines": ["Warning: /etc/puppet/hiera.yaml: Use of 'hiera.yaml' version 3 is deprecated. It should be converted to version 5", "   (file: /etc/puppet/hiera.yaml)", "Warning: Undefined variable '::deploy_config_name'; ", "   (file & line not available)", "Warning: The function 'hiera' is deprecated in favor of using 'lookup'. See https://puppet.com/docs/puppet/7.10/deprecated_language.html", "   (file & line not available)", "Deprecation Warning: This command is deprecated and will be removed. Please use 'pcs property config' instead.", "Deprecation Warning: This command is deprecated and will be removed. Please use 'pcs constraint config' instead.", "Deprecation Warning: This command is deprecated and will be removed. Please use 'pcs constraint config' instead.", "Error: pcs -f /var/lib/pacemaker/cib/puppet-cib-backup20230606-143531-1pzw1f constraint location galera-bundle rule resource-discovery=exclusive score=0 galera-role eq true failed: Error: duplicate constraint already exists, use --force to override. Too many tries", "Error: /Stage[main]/Tripleo::Profile::Pacemaker::Database::Mysql_bundle/Pacemaker::Resource::Ocf[galera]/Pcmk_resource[galera]/ensure: change from 'absent' to 'present' failed: pcs -f /var/lib/pacemaker/cib/puppet-cib-backup20230606-143531-1pzw1f constraint location galera-bundle rule resource-discovery=exclusive score=0 galera-role eq true failed: Error: duplicate constraint already exists, use --force to override. Too many tries"], "stdout": "Notice: Compiled catalog for standalone.localdomain in environment production in 1.65 seconds\nNotice: /Stage[main]/Pacemaker::Stonith/Pacemaker::Property[Disable STONITH]/Pcmk_property[property--stonith-enabled]/ensure: created\nNotice: /Stage[main]/Tripleo::Profile::Pacemaker::Database::Mysql_bundle/Pacemaker::Property[galera-role-standalone]/Pcmk_property[property-standalone-galera-role]/ensure: created\nNotice: /Stage[main]/Tripleo::Profile::Pacemaker::Database::Mysql_bundle/Pacemaker::Resource::Bundle[galera-bundle]/Pcmk_bundle[galera-bundle]/ensure: created\nNotice: Applied catalog in 58.37 seconds\nApplication:\n   Initial environment: production\n   Converged environment: production\n         Run mode: user\nChanges:\n            Total: 3\nEvents:\n          Failure: 1\n          Success: 3\n            Total: 4\nResources:\n           Failed: 1\n          Skipped: 141\n          Changed: 3\n      Out of sync: 4\n            Total: 145\nTime:\n   Config retrieval: 1.85\n         Last run: 1686054888\n    Pcmk resource: 42.45\n   Transaction evaluation: 58.36\n   Catalog application: 58.37\n    Pcmk property: 6.69\n      Pcmk bundle: 8.98\n            Total: 58.37\nVersion:\n           Config: 1686054828\n           Puppet: 7.10.0", "stdout_lines": ["Notice: Compiled catalog for standalone.localdomain in environment production in 1.65 seconds", "Notice: /Stage[main]/Pacemaker::Stonith/Pacemaker::Property[Disable STONITH]/Pcmk_property[property--stonith-enabled]/ensure: created", "Notice: /Stage[main]/Tripleo::Profile::Pacemaker::Database::Mysql_bundle/Pacemaker::Property[galera-role-standalone]/Pcmk_property[property-standalone-galera-role]/ensure: created", "Notice: /Stage[main]/Tripleo::Profile::Pacemaker::Database::Mysql_bundle/Pacemaker::Resource::Bundle[galera-bundle]/Pcmk_bundle[galera-bundle]/ensure: created", "Notice: Applied catalog in 58.37 seconds", "Application:", "   Initial environment: production", "   Converged environment: production", "         Run mode: user", "Changes:", "            Total: 3", "Events:", "          Failure: 1", "          Success: 3", "            Total: 4", "Resources:", "           Failed: 1", "          Skipped: 141", "          Changed: 3", "      Out of sync: 4", "            Total: 145", "Time:", "   Config retrieval: 1.85", "         Last run: 1686054888", "    Pcmk resource: 42.45", "   Transaction evaluation: 58.36", "   Catalog application: 58.37", "    Pcmk property: 6.69", "      Pcmk bundle: 8.98", "            Total: 58.37", "Version:", "           Config: 1686054828", "           Puppet: 7.10.0"]}
```

**Comment by jistr** [comment]
Rebased onto master after #52 merge.

**Comment by jistr** [comment]
recheck

**Comment by jistr** [comment]
recheck

**Comment by jistr** [comment]
```
TASK [backend_services : Patch rabbitmq resources for lower resource consumption] ***
fatal: [localhost]: FAILED! => {"changed": false, "cmd": "crname=$(oc get openstackcontrolplane -o name)\noc patch ${crname} --type json  -p='[{\"op\": \"replace\", \"path\": \"/spec/rabbitmq/templates/rabbitmq/resources/requests/cpu\", \"value\": 500m}]'\noc patch ${crname} --type json  -p='[{\"op\": \"replace\", \"path\": \"/spec/rabbitmq/templates/rabbitmq/resources/requests/memory\", \"value\": 500Mi}]'\noc patch ${crname} --type json  -p='[{\"op\": \"replace\", \"path\": \"/spec/rabbitmq/templates/rabbitmq-cell1/resources/requests/cpu\", \"value\": 500m}]'\noc patch ${crname} --type json  -p='[{\"op\": \"replace\", \"path\": \"/spec/rabbitmq/templates/rabbitmq-cell1/resources/requests/memory\", \"value\": 500Mi}]'\n", "delta": "0:00:17.600244", "end": "2023-06-09 07:37:10.219531", "msg": "non-zero return code", "rc": 1, "start": "2023-06-09 07:36:52.619287", "stderr": "error: You must be logged in to the server (Unauthorized)\nerror: You must be logged in to the server (Unauthorized)\nerror: You must be logged in to the server (Unauthorized)", "stderr_lines": ["error: You must be logged in to the server (Unauthorized)", "error: You must be logged in to the server (Unauthorized)", "error: You must be logged in to the server (Unauthorized)"], "stdout": "openstackcontrolplane.core.openstack.org/openstack patched", "stdout_lines": ["openstackcontrolplane.core.openstack.org/openstack patched"]}
```

Huh i guess i'll wait a bit and recheck again.

**Comment by jistr** [comment]
recheck

**Comment by jistr** [comment]
CI green :green_circle:

File: docs/openstack/backend_services_deployment.md [hunk 3]
```diff
@@ -103,40 +104,68 @@ podified OpenStack control plane services.
...
+
+    horizon:
+      enabled: false
+      template: {}
+
+    ironic:
+      enabled: false
+      template:
+        ironicConductors: []
+
```
Comment (fao89) [inline docs/openstack/backend_services_deployment.md:142]: I believe we need more updates for network isolation: https://github.com/openstack-k8s-operators/openstack-operator/blob/main/config/samples/core_v1beta1_openstackcontrolplane_network_isolation.yaml

File: docs/openstack/mariadb_copy.md [hunk 4]
```diff
@@ -33,12 +33,12 @@ PODIFIED_MARIADB_IP=$(oc get -o yaml pod mariadb-openstack | grep podIP: | awk '
...
 
 # Use your environment's values for these:
-EXTERNAL_MARIADB_IP=192.168.24.3
+EXTERNAL_MARIADB_IP=192.168.122.100
 EXTERNAL_DB_ROOT_PASSWORD=$(cat ~/tripleo-standalone-passwords.yaml | grep ' MysqlRootPassword:' | awk -F ': ' '{ print $2; }')
 PODIFIED_DB_ROOT_PASSWORD=12345678
 
 # ssh commands to reach controller machines
-CONTROLLER1_SSH="ssh -F ~/director_standalone/vagrant_ssh_config vagrant@standalone"
+CONTROLLER1_SSH="ssh -i ~/install_yamls/out/edpm/ansibleee-ssh-key-id_rsa root@192.168.122.100"
```
Comment (fao89) [inline docs/openstack/mariadb_copy.md:41]: 👍🏽

**Review by fao89 — APPROVED** [review]
added a suggestion about the ctlplane CR, but I'm good with merging this as the CI is happy

File: docs/openstack/backend_services_deployment.md (hunk 3 above)
Comment (jistr) [reply docs/openstack/backend_services_deployment.md:142]: Yea that's pretty much where i'm taking the info from, but here for the "backend services" section we only include parameters for the services that are enabled. So e.g. network isolation parameters for Keystone went into the `keystone_adoption.md`, and i have WIP updates for the other services that i'll post soon.

**Comment by jistr** [comment]
Thanks, merging :rocket:

**Pull request merged**